src/auth0-session/client/edge-client.ts

```diff
@@ -0,0 +1,241 @@
+    extras: CallbackExtras
+  ): Promise<TokenEndpointResponse> {
+    const [as, client] = await this.getClient();
+
+    const { clientAssertionSigningKey, clientAssertionSigningAlg } = this.config;
+
+    let clientPrivateKey = clientAssertionSigningKey as CryptoKey | undefined;
+    /* c8 ignore next 3 */
+    if (clientPrivateKey && !(clientPrivateKey instanceof CryptoKey)) {
+      clientPrivateKey = await jose.importPKCS8<CryptoKey>(clientPrivateKey, clientAssertionSigningAlg || 'RS256');
+    }
+    const response = await oauth.authorizationCodeGrantRequest(
+      as,
+      client,
+      parameters,
+      redirectUri,
+      checks.code_verifier as string,
+      {
+        additionalParameters: extras.exchangeBody,
+        ...(clientPrivateKey && { clientPrivateKey }),
+        ...this.httpOptions()
+      }
+    );
+
+    const result = await oauth.processAuthorizationCodeOpenIDResponse(
+      as,
+      client,
+      response,
+      checks.nonce,
+      checks.max_age
+    );
+    if (oauth.isOAuth2Error(result)) {
+      throw new IdentityProviderError({
+        message: result.error_description || /* c8 ignore next  */ result.error,
+        error: result.error,
+        error_description: result.error_description
+      });
+    }
+    return result;
+  }
+
+  async endSessionUrl(parameters: EndSessionParameters): Promise<string> {
+    const [as] = await this.getClient();
+    const issuerUrl = new URL(as.issuer);
+
+    if (
+      this.config.idpLogout &&
+      (this.config.auth0Logout || (issuerUrl.hostname.match('\\.auth0\\.com$') && this.config.auth0Logout !== false))
+    ) {
+      const { id_token_hint, post_logout_redirect_uri, ...extraParams } = parameters;
+      const auth0LogoutUrl: URL = new URL(urlJoin(as.issuer, '/v2/logout'));
+      post_logout_redirect_uri && auth0LogoutUrl.searchParams.set('returnTo', post_logout_redirect_uri);
+      auth0LogoutUrl.searchParams.set('client_id', this.config.clientID);
+      Object.entries(extraParams).forEach(([key, value]: [string, string]) => {
+        if (value === null || value === undefined) {
+          return;
+        }
+        auth0LogoutUrl.searchParams.set(key, value);
+      });
+      return auth0LogoutUrl.toString();
+    }
+    if (!as.end_session_endpoint) {
+      throw new Error('no rp inititated logout');
+    }
+    const oidcLogoutUrl = new URL(as.end_session_endpoint);
+    Object.entries(parameters).forEach(([key, value]: [string, string]) => {
+      if (value === null || value === undefined) {
+        return;
+      }
+      oidcLogoutUrl.searchParams.set(key, value);
+    });
+
+    oidcLogoutUrl.searchParams.set('client_id', this.config.clientID);
+    return oidcLogoutUrl.toString();
+  }
+
+  async userinfo(accessToken: string): Promise<Record<string, unknown>> {
+    const [as, client] = await this.getClient();
+    const response = await oauth.userInfoRequest(as, client, accessToken, this.httpOptions());
```

Comment:
Can this call throw an exception? If so, should it be caught, wrapped, and rethrown?

Reply:
This get's caught here https://github.com/auth0/nextjs-auth0/blob/edge-auth-handlers/src/handlers/profile.ts#L237

Reply:
In the node handler we're wrapping the errors in our own, more specific error types: https://github.com/auth0/nextjs-auth0/blob/141ec7bff57684747e99baddf02840706adda33b/src/auth0-session/client/node-client.ts#L188

Reply:
Not for `userinfo` - but yeah - fair enough, I'll create a `UserInfoError` and add it to the node and edge clients